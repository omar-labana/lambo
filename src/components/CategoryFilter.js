import PropTypes from 'prop-types';

const categories = ['All', 'Chicken', 'Pasta', 'Seafood'];

const CategoryFilter = (props) => {
  const { target, handleFilter } = props;

  return (
    <div className="flex items-center justify-center gap-5 py-10">
      <h2 className="text-white text-3xl">Category Filter:</h2>
      <select name="target" value={target} onChange={handleFilter} className="p-5 text-3xl">
        {categories.map((category) => (
          <option key={category} value={category}>
            {category}
          </option>
        ))}
      </select>
    </div>
  );
};

CategoryFilter.propTypes = {
  target: PropTypes.string.isRequired,
  handleFilter: PropTypes.func.isRequired,
};

export default CategoryFilter;
