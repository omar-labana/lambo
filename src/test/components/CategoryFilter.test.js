import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import CategoryFilter from '../../components/CategoryFilter';

describe('CategoryFilter Component', () => {
  const target = 'ALL';
  const handleFilter = () => '1';
  const wrapper = shallow(<CategoryFilter target={target} handleFilter={handleFilter} />);

  it('exists', () => {
    expect(wrapper.find('CategoryFilter')).toBeTruthy();
  });

  it('has to match snap', () => {
    const home = renderer.create(<CategoryFilter target={target} handleFilter={handleFilter} />);
    expect(home).toMatchSnapshot();
  });
});
