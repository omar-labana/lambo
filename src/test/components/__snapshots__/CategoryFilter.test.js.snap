// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryFilter Component has to match snap 1`] = `
<div
  className="flex items-center justify-center gap-5 py-10"
>
  <h2
    className="text-white text-3xl"
  >
    Category Filter:
  </h2>
  <select
    className="p-5 text-3xl"
    name="target"
    onChange={[Function]}
    value="ALL"
  >
    <option
      value="All"
    >
      All
    </option>
    <option
      value="Chicken"
    >
      Chicken
    </option>
    <option
      value="Pasta"
    >
      Pasta
    </option>
    <option
      value="Seafood"
    >
      Seafood
    </option>
  </select>
</div>
`;
